test/integration/command-logging-and-monitoring/command_logging_and_monitoring.prose.test.ts

```diff
@@ -0,0 +1,238 @@
+        the "command" value is a string of length 5 + (length of trailing ellipsis).
+
+    4. Inspect the resulting "command succeeded" log message and assert that
+        the "reply" value is a string of length 5 + (length of trailing ellipsis).
+        
+    5. (Optional)
+        If the driver attaches raw server responses to failures
+        and can access these via log messages to assert on, 
+        run the command {"notARealCommand": true}. 
+
+        Inspect the resulting "command failed" log message 
+        and confirm that the server error is a string of length 5 + (length of trailing ellipsis).
+    */
+    it('should follow custom truncation limit', {
+      metadata: {},
+      test: async function () {
+        const writable = {
+          buffer: [],
+          write(log) {
+            this.buffer.push(log);
+          }
+        };
+
+        // 1.
+        const client = this.configuration.newClient(
+          {},
+          {
+            [loggerFeatureFlag]: true,
+            mongodbLogPath: writable,
+            mongodbLogComponentSeverities: {
+              command: 'debug'
+            },
+            mongodbLogMaxDocumentLength: 5
+          }
+        );
+
+        // 2.
+        await client.db('admin').command({ hello: 'true' });
+
+        // remove sensitive commands for uniformity
+        const cleanedBuffer = [];
+        for (let i = 0; i < writable.buffer.length; i++) {
+          if (writable.buffer[i].command !== '{}' && writable.buffer[i].reply !== '{}') {
+            cleanedBuffer.push(writable.buffer[i]);
+          }
+        }
+
+        // 3.
+        const insertManyCommandStarted = cleanedBuffer[0];
+        expect(insertManyCommandStarted?.message).to.equal('Command started');
+        expect(insertManyCommandStarted?.command).to.be.a('string');
+        expect(insertManyCommandStarted?.command?.length).to.equal(5 + ELLIPSES_LENGTH);
+
+        // 4.
+        const insertManyCommandSucceeded = cleanedBuffer[1];
+        expect(insertManyCommandSucceeded?.message).to.equal('Command succeeded');
+        expect(insertManyCommandSucceeded?.reply).to.be.a('string');
+        expect(insertManyCommandSucceeded?.reply?.length).to.be.at.most(5 + ELLIPSES_LENGTH);
+
+        await client.close();
+      }
+    });
+  });
+  context('Truncation with multi-byte codepoints', function () {
+    /*
+    A specific test case is not provided here due to the allowed variations in truncation logic
+     as well as varying extended JSON whitespace usage. 
+     Drivers MUST write language-specific tests that confirm truncation of commands, replies, 
+     and (if applicable) server responses included in error messages 
+     work as expected when the data being truncated includes multi-byte Unicode codepoints.
+     
+     If the driver uses anything other than Unicode codepoints as the unit for max document length, 
+     there also MUST be tests confirming that cases
```

Comment:
Not sure what may be going on with the linter in CI, but on my machine prettier has a bunch of fixes to this file, like trailing whitespace fixes.

Reply:
I think the linter problem was a stray only / smg along those lines. My machine doesn't have any trailing whitespace fixes on this file.